imap_processing/glows/l0/glows_l0_data.py

```diff
@@ -0,0 +1,153 @@
+class GlowsHistL0:
+    """Data structure for storing GLOWS histogram packet data.
+    Parameters
+    ----------
+    packet : tuple[list]
+        Histogram packet yielded from space_packet_parser.generate_packets
+
+    Attributes
+    ----------
+    L0_KEYS : tuple[str]
+        Data names from the packet
+    data: dict
+        The data from the packet
+
+    Methods
+    -------
+    process_packet(packet)
+        Process the given packet into the data dictionary
+
+    _convert_histogram_data(binary_hist_data)
+        Process the binary histogram data into 8-bit bins, in a list.
+    """
+
+    L0_KEYS: tuple[str] = (
+        "SHCOARSE",
+        "STARTID",
+        "ENDID",
+        "FLAGS",
+        "SWVER",
+        "SEC",
+        "SUBSEC",
+        "OFFSETSEC",
+        "OFFSETSUBSEC",
+        "GLXSEC",
+        "GLXSUBSEC",
+        "GLXOFFSEC",
+        "GLXOFFSUBSEC",
+        "SPINS",
+        "NBINS",
+        "TEMPAVG",
+        "TEMPVAR",
+        "HVAVG",
+        "HVVAR",
+        "SPAVG",
+        "SPVAR",
+        "ELAVG",
+        "ELVAR",
+        "EVENTS",
+        "HISTOGRAM_DATA",
+    )
+
+    def __repr__(self):
+        return str(self.data)
+
+    def __init__(self, packet):
+        self.data = self.process_packet(packet)
+
+    def process_packet(self, packet) -> dict:
+        """Process the GLOWS histogram packet, which is yielded from space_packet_parser
+
+        Parameters
+        ----------
+        packet
+            Histogram packet from space_packet_parser
+        Returns
+        -------
+        data: dict
+            Data dictionary created from the packet, using L0_KEYS
+        """
+        data = {}
+        for key in self.L0_KEYS:
+            if key != "HISTOGRAM_DATA":
+                data[key] = packet.data[key].derived_value
+            else:
+                data[key] = self._convert_histogram_data(packet.data[key].raw_value)
+            if data[key] is None:
+                raise ValueError(f"Missing data for {key} in {packet.data}")
+        return data
+
+    def _convert_histogram_data(self, binary_hist_data: str) -> list[int]:
```

Comment:
can we replace this function with this line? I did this in SWE. SWE also needed to get data in 8-bit chunks too.
```
binary_hist_data = packet.data["HISTOGRAM_DATA"].raw_value
byte_data = int(binary_hist_data, 2).to_bytes(3599, byteorder="big")
# convert bytes to numpy array of uint8
raw_counts = np.frombuffer(byte_data, dtype=np.uint8)
```

Reply:
No, that doesn't work. I get an overflow error from trying to convert the binary_hist_data to an int.

Reply:
bummer. ok